### PR TITLE
Don't show the currently-completing thing at the cursor in JS files

### LINKED ISSUE #6693: Salsa: bogus intelli-sense proposal
_From @isidorn on January 26, 2016 16:20_

* Have the following code:
```
Person.prototype.getName5 = function getName() {
  return this.name;
};

Person.getNa
```
* Trigger intelli-sense on the last line, last column, getNa*
* The first two proposals are bogus and should not appear.

screen shot 2016-01-26 at 17 16 27

_Copied from original issue: Microsoft/vscode#2386_

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #6693

### CODE REVIEW

**Comment by mhegazy** [comment]
nice :wink:

**Comment by mhegazy** [comment]
:+1:

**Comment by vladima** [comment]
:+1:

**Comment by RyanCavanaugh** [comment]
Investigating test failure...

File: src/services/services.ts [hunk 1]
```diff
@@ -7530,7 +7533,7 @@ namespace ts {
         function walk(node: Node) {
             switch (node.kind) {
                 case SyntaxKind.Identifier:
-                    nameTable[(<Identifier>node).text] = (<Identifier>node).text;
+                    nameTable[(<Identifier>node).text] = nameTable[(<Identifier>node).text] === undefined ? node.pos : -1;
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: Isn't this too order-dependent? What happens if you add code *prior* to another usage? For instance, I start out with:

```ts
if (Person.blah) {
    doSomething();
}
```

I'd want to also get completion for if I added a use of `Person.blah` before the `if` statement:

```
Person.blah/**/

if (Person.blah) {
    doSomething();
}
```

Comment (RyanCavanaugh) [inline src/services/services.ts]: If you have multiple uses, the value in the table becomes `-1` and no filtering occurs

Comment (DanielRosenwasser) [inline src/services/services.ts]: Ah, okay. I misunderstood.

Why don't you use an enum that would make this explicit?

```ts
const name = (node as Identifier).text;
nameTable[name] = hasProperty(nameTable, name) ? NameTableState.Once : NameTableState.Many;
```

or at the very least, leave a nice big comment explaining what you're doing here and what `-1` means.

Comment (RyanCavanaugh) [inline src/services/services.ts]: Because we need the position in the case where it's Once.

Comment (DanielRosenwasser) [inline src/services/services.ts]: Ah, right. You're not checking the current identifier and asking "should I actually show this", you're just grabbing up identifiers. I'm not sure why I missed that.

**Pull request merged**